### PR TITLE
ddl: fix a bug that rollback distributed adding index is not handled correctly

### LINKED ISSUE #43915: rollback distributed adding index is not handled correctly.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

set global tidb_enable_dist_task=1;
creating an index on a large table and cancel it.

### 2. What did you expect to see? (Required)

The next add index SQL will use ingest way to add index again.

### 3. What did you see instead (Required)

The next add index SQL won't use ingest way to add index.

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #43915

Problem Summary:

### What is changed and how it works?
Unregister the lightning backend when rollback.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/scheduler.go [hunk 1]
```diff
@@ -281,6 +291,14 @@ func (b *backfillSchedulerHandle) CleanupSubtaskExecEnv(context.Context) error {
 
 // Rollback implements the Scheduler interface.
 func (b *backfillSchedulerHandle) Rollback(context.Context) error {
+	logutil.BgLogger().Info("[ddl] rollback backfill add index task")
+	bc, err := ingest.LitBackCtxMgr.Register(b.d.ctx, b.index.Unique, b.job.ID)
```
Comment (lcwangchao) [inline ddl/scheduler.go:295]: Why `ingest.LitBackCtxMgr.Register` should be called here?

**Review by lcwangchao — APPROVED** [review]

File: ddl/scheduler.go (hunk 1 above)
Comment (tangenta) [reply ddl/scheduler.go:295]: +1. I suggest to use `ingest.LitBackCtxMgr.Load()` instead.

File: ddl/scheduler.go [hunk 2]
```diff
@@ -281,6 +291,11 @@ func (b *backfillSchedulerHandle) CleanupSubtaskExecEnv(context.Context) error {
 
 // Rollback implements the Scheduler interface.
 func (b *backfillSchedulerHandle) Rollback(context.Context) error {
+	logutil.BgLogger().Info("[ddl] rollback backfill add index task")
```
Comment (tangenta) [inline ddl/scheduler.go:294]: Please add job ID info to make it easier to debug.

**Review by tangenta — APPROVED** [review]

**Pull request merged**